checkov/arm/checks/resource/AppServiceHttps20Enabled.py

```diff
@@ -1,23 +1,33 @@
+from __future__ import annotations
+
 from typing import Any
 
-from checkov.arm.base_resource_value_check import BaseResourceValueCheck
-from checkov.common.models.enums import CheckCategories
+from checkov.arm.base_resource_check import BaseResourceCheck
+from checkov.common.models.enums import CheckResult, CheckCategories
 
 
-class AppServiceHttps20Enabled(BaseResourceValueCheck):
+class AppServiceHttps20Enabled(BaseResourceCheck):
+    # apiVersion = 2018-11-01 - http20Enabled is a string
+    # apiVersion > 2020-10-01  - http20Enabled is a boolean
     def __init__(self) -> None:
         name = "Ensure that 'HTTP Version' is the latest if used to run the web app"
         id = "CKV_AZURE_18"
         supported_resources = ("Microsoft.Web/sites",)
         categories = (CheckCategories.NETWORKING,)
         super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
 
-    def get_inspected_key(self) -> str:
-        # https://docs.microsoft.com/en-us/azure/templates/microsoft.web/2019-08-01/sites
-        return "properties/siteConfig/http20Enabled"
-
-    def get_expected_value(self) -> Any:
-        return "true"
+    def scan_resource_conf(self, conf: dict[str, Any]) -> CheckResult:
+        if "properties" in conf:
+            if "siteConfig" in conf["properties"] and conf["properties"]["siteConfig"]:
+                if "http20Enabled" in conf["properties"]["siteConfig"]:
+                    http_20_enabled = conf["properties"]["siteConfig"]["http20Enabled"]
```

Comment:
you could use `find_in_dict()` like this to do the same 🙂 
```suggestion
        http_20_enabled = find_in_dict(conf, "properties/siteConfig/http20Enabled")
```

Reply:
@gruebel Nice one